cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -512,11 +513,13 @@ protected PersistentVolumeClaim createPersistentVolumeClaim(String name) {
                 .withRequests(requests)
                 .endResources()
                 .withStorageClassName(storage.getStorageClass())
-                .withNewSelector().withMatchLabels(storage.getSelector()).endSelector()
-                .endSpec()
-                .build();
+                .endSpec();
+
+        if (!storage.getSelector().isEmpty()) {
+            pvcb.editSpec().withSelector(new LabelSelector(null, storage.getSelector())).endSpec();
```

Comment:
I think it would be more consistent with the other code in `AbstractModel` to create the selector before we instantiate the `PersistentVolumeClaimBuilder` (with a null selector if the `storage` has no selector) and then just call `.withSelector()` when using the `PersistentVolumeClaimBuilder`. The `edit*()` methods are great when you have an object whose creation you can't change (as in the tests), but that's not the case here.

Reply:
Makes sense, done!